tests/test_widget.py

```diff
@@ -26,3 +32,29 @@ def test_widget_set_visible_invalid_string():
         widget.visible = "nope! no widget for me!"
 
     assert widget.visible
+
+
+@pytest.mark.parametrize(
+    "size_input, expectation",
+    [
+        (None, does_not_raise()),
+        (10, does_not_raise()),
+        (10.0, does_not_raise()),
+        (10.2, does_not_raise()),
+        (Scalar(100, Unit.CELLS, Unit.WIDTH), does_not_raise()),
+        (Scalar(10.2, Unit.CELLS, Unit.WIDTH), does_not_raise()),
+        ("10", does_not_raise()),
+        # And now for some common types we don't handle...
+        ("a", pytest.raises(StyleValueError)),
+        (list(), pytest.raises(StyleValueError)),
+        (tuple(), pytest.raises(StyleValueError)),
+        (dict(), pytest.raises(StyleValueError)),
+        (3.14j, pytest.raises(StyleValueError)),
+        (Decimal("3.14"), pytest.raises(StyleValueError)),
+    ],
+)
+def test_widget_style_size_can_accept_various_data_types(size_input, expectation):
```

Comment:
I'd say we want to do more than just check that this doesn't raise an exception. For each supported datatype, we should assert that the value we read back from the property is what we would expect.

For example, if I do `widget.styles.width = 10` vs `widget.styles.width = 10.0`, then read `widget.styles.width`, will I get the same value back?

Personally I would keep this test for just the common datatypes we don't support, and have separate tests for checking we write and read the expected values for different types.

I'd put this in `test_styles` rather than `test_widget` too, since that file already contains test of that sort.

Reply:
Agree that it probably does belong in `test_styles.py`. 

Suggest we flesh out tests like this in the near future, when the Styles object is closer to being finalized.